DigitalLearningSolutions.Web.Tests/Controllers/Support/FaqControllerTests.cs

```diff
@@ -0,0 +1,193 @@
+            faqsService = A.Fake<IFaqsService>();
+            A.CallTo(() => faqsService.GetPublishedFaqsForTargetGroup(0))
+                .Returns(faqs);
+            A.CallTo(() => faqsService.GetPublishedFaqByIdForTargetGroup(2, 0))
+                .Returns(faqs.ElementAt(1));
+            A.CallTo(() => faqsService.GetPublishedFaqByIdForTargetGroup(5, 0))
+                .Returns(null);
+            A.CallTo(() => featureManager.IsEnabledAsync(FeatureFlags.RefactoredTrackingSystem))
+                .Returns(true);
+        }
+
+        [Test]
+        public async Task TrackingSystem_Faqs_page_should_be_shown_for_valid_claims()
+        {
+            // Given
+            var controller = new FaqsController(featureManager, configuration, faqsService)
+                .WithDefaultContext()
+                .WithMockUser(true, isCentreAdmin: true);
+
+            // When
+            var result = await controller.Index(DlsSubApplication.TrackingSystem);
+
+            // Then
+            result.Should().BeViewResult().WithViewName("Faqs");
+        }
+
+        [Test]
+        public async Task Frameworks_Faqs_page_should_be_shown_for_valid_claims()
+        {
+            // Given
+            var controller = new FaqsController(featureManager, configuration, faqsService)
+                .WithDefaultContext()
+                .WithMockUser(true, isCentreAdmin: false, isFrameworkDeveloper: true);
+
+            // When
+            var result = await controller.Index(DlsSubApplication.Frameworks);
+
+            // Then
+            result.Should().BeViewResult().WithViewName("Faqs");
+        }
+
+        [Test]
+        public async Task Invalid_application_name_should_redirect_to_404_page()
+        {
+            // Given
+            var controller = new FaqsController(featureManager, configuration, faqsService)
+                .WithDefaultContext()
+                .WithMockUser(true, isCentreAdmin: true, isFrameworkDeveloper: true);
+
+            // When
+            var result = await controller.Index(DlsSubApplication.Supervisor);
+
+            // Then
+            result.Should().BeNotFoundResult();
+        }
+
+        [Test]
+        public async Task Home_page_should_be_shown_when_accessing_tracking_system_faqs_without_appropriate_claims()
+        {
+            // Given
+            var controller = new FaqsController(featureManager, configuration, faqsService)
+                .WithDefaultContext()
+                .WithMockUser(true, isCentreAdmin: false, isFrameworkDeveloper: true);
+
+            // When
+            var result = await controller.Index(DlsSubApplication.TrackingSystem);
+
+            // Then
+            result.Should().BeRedirectToActionResult().WithControllerName("Home").WithActionName("Index");
+        }
+
+        [Test]
+        public async Task
+            Home_page_should_be_shown_when_accessing_tracking_system_with_refactored_tracking_system_disabled()

```

Comment:
Really I would expect this to give you a Not Found rather than sending you to home. I think that's what the ValidateAllowedDlsSubApplication type filter does anyway.

Reply:
To be honest this was taken from another implementation. But all gone now that the `ValidateAllowedDlsSubApplication` is applied.